File: 713-subarray-product-less-than-k/713-subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
            if k <= 1 :
                 return 0
            ans = 0
            mul = 1
            l  ,r = 0 , 0
            while r < len(nums):
                mul *= nums[r]
                while mul >= k :
                    mul /= nums[l]
                    l+=1
                ans += r - l + 1
                r += 1
            return  ans

File: 713-subarray-product-less-than-k/713-subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        if (k <= 1 )return 0 ;

	int n = (int)nums.size();
	int mul = 1 ;
	int l = 0 , r = 0  ;
	int ans = 0 ;
	while(r < n ){
		mul *= nums[r];
		while(mul >= k){
			mul /= nums[l];
			l++;
		}
		ans += (r - l + 1);
		r++;
	}

	return ans;
    }
};